### ISSUE TITLE
Lexer always use latest PHP version

### ISSUE THREAD

**Issue by MatTheCat** [composer-unused/composer-unused#292]
## Describe the bug
I just tried composer-unused with PHP 7.4 and got

> Syntax error, unexpected T_MATCH on line 43

This is because the project I’m working on includes PHPUnit 7.5.8 which defines [a `Match` interface](https://github.com/sebastianbergmann/phpunit/blob/7.5.8/src/Framework/MockObject/Builder/Match.php). The default lexer is the `Emulative` one [for the newest PHP version](https://github.com/nikic/PHP-Parser/blob/a6e34665fd6a2bcefc924cb5fd73788767ae510e/lib/PhpParser/Lexer/Emulative.php#L40) which is now PHP 8.1, so it reports an error when seeing the `Match` interface because `match` is then considered a keyword.

I think we want to configure the parser with the PHP version in use, in which case we need to pass it the default `Lexer` (cf. https://github.com/nikic/PHP-Parser/issues/690#issuecomment-671468752).

**Comment by icanhazstring** [comment]
Hi. Thanks for report. Did you run the phar bundle? This should run fine.

Local installation is always tricky.

**Comment by MatTheCat** [comment]
Same issue running the PHAR: the code is parsed as the latest PHP version, which is wrong.

**Comment by icanhazstring** [comment]
Ok, thanks for the clarification.
Will try to come up with a test case to actually test this and implement a configuration or extrapolate the PHP version from the projects `composer.json`

**Comment by MatTheCat** [comment]
> Will try to come up with a test case to actually test this

I think running `composer-unused` against a project containing a `Match` class or interface would be enough, as it would fail using PHP7 :thinking: 

> implement a configuration or extrapolate the PHP version from the projects `composer.json`

Wouldn’t it be simpler to parse the code as the running PHP version? All it would take is a second argument to `PhpParser\ParserFactory::create`:

```diff
+ use PhpParser\Parser\Tokens\Lexer;

- (new ParserFactory())->create(ParserFactory::ONLY_PHP7)
+ (new ParserFactory())->create(ParserFactory::ONLY_PHP7, new Lexer())
```

**Comment by icanhazstring** [comment]
Yep. I implemented a fix using the current `PHP_VERSION_ID` to determine the lexer.
I needed to rework the dependency injection quite a bit to make it easier for me. 

But the line you proposed is pretty much the fix :)

**Comment by icanhazstring** [comment]
Solved in `dev-main` coming soon with `8.1` :)
